db/schema_tables.cc

```diff
@@ -1774,7 +1780,11 @@ static shared_ptr<cql3::functions::user_aggregate> create_aggregate(replica::dat
 
     bytes_opt initcond = std::nullopt;
     if (initcond_str) {
-        initcond = state_type->from_string(initcond_str.value());
+        auto expr = cql3::util::do_with_parser(*initcond_str, std::mem_fn(&cql3_parser::CqlParser::term));
+        auto dummy_ident = ::make_shared<cql3::column_identifier>("", true);
+        auto column_spec = make_lw_shared<cql3::column_specification>("", "", dummy_ident, state_type);
```

Comment:
Why is this "dummy column spec" thing needed? Can't prepare_expression deal with none? (maybe this is a stupid question, but I'm not familiar enough with the expression evaluation code).

Reply:
The "dummy column spec" is currently needed to give the parser the type of the expression. Perhaps  we can drop this in the future, because we have a few TODOs for inferring types: https://github.com/scylladb/scylladb/blob/3ffc978166abd50e7de545a8441cac34c0ea1ff0/cql3/expr/prepare_expr.cc#L335-L337
The "dummy column ident" is also needed because we're using its value in the implementation (although it can be any value in this case)

But the main reason it's implemented like this is to mirror the implementation used for parsing the incoming cql requests (the change only concerns parsing of the initcond of an aggregate that is already in the `aggregates` table)
https://github.com/scylladb/scylladb/blob/3ffc978166abd50e7de545a8441cac34c0ea1ff0/cql3/statements/create_aggregate_statement.cc#L66-L69

Reply:
Ok. We could have moved this ugliness into a variant of prepare_expression() which takes only a type (not a column spec) and does this dummy-column-spec song and dance, but seeing that you copied a trick already used elsewhere, I guess no harm in having another one.

Reply:
Type inference won't help here because you can't infer the type from a constant.